src/tcgwars/logic/impl/gen8/BattleStyles.groovy

```diff
@@ -998,13 +998,18 @@ public enum BattleStyles implements LogicCardInfo {
           energyCost COLORLESS, COLORLESS
           attackRequirement {}
           onAttack {
-            // TODO
             damage 10
-            def tar = opp.deck.subList(0,6).filterByType(ITEM)
 
-            my.hand.filterByType(ITEM)
-            if(tar){
-              tar.select(max:tar.size(),"Choose the item to discard").discard()
+            def ballCards = my.hand.filterByType(ITEM).filterByNameLike("Ball")
```

Comment:
Make sure that this doesn't match Bursting *Ball*oon, or *Ball*oon Berry. I think most relevant cards end with "Ball"

Reply:
Relevant rulings: https://compendium.pokegym.net/category/5-trainers/apricorn-maker/

Reply:
Updated the code to check for cards that end in "Ball"